homeassistant/components/sensor/waze_travel_time.py

```diff
@@ -0,0 +1,177 @@
+            )
+
+
+class WazeTravelTime(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, waze_data, output_type, update_interval, name):
+        """Initialize the sensor."""
+        self.type = output_type
+        self.sensor_name = name
+        self._name = OUTPUT_TYPES[output_type][0]
+        self._unit_of_measurement = OUTPUT_TYPES[output_type][1]
+        self._icon = OUTPUT_TYPES[output_type][2]
+        self._resolution = OUTPUT_TYPES[output_type][3]
+        self.waze_data = waze_data
+        self._state = None
+        self.update = Throttle(update_interval)(self.__update)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.sensor_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    def __update(self):
+        """Fetch new state data for the sensor.
+
+        This is the only method that should fetch new data for Home Assistant.
+        """
+        try:
+            self.waze_data.update()
+            if self.type in self.waze_data.data:
+                if self.type == 'route':
+                    self._state = self.waze_data.data[self.type]
+                    _LOGGER.debug(
+                        "Retreived route is %s = %s %s",
+                        self.type,
+                        self._state
+                        )
+                else:
+                    self._state = round(
+                        self.waze_data.data[self.type], self._resolution)
+                    _LOGGER.debug(
+                        "Retreived route is %s = %s %s",
+                        self.type,
+                        self._state,
+                        self._unit_of_measurement
+                        )
+        except KeyError:
+            _LOGGER.error("Error retreiving data from server")
+
+
+class WazeRouteData(object):
+    """Get data from WazeRouteCalculator."""
+
+    def __init__(self, origin, destination, region, update_interval):
+        """Initialize the data object."""
+        self._origin = origin
+        self._destination = destination
+        self._region = region
+        self.data = {}
+
+        self.update = Throttle(update_interval)(self.__update)
+
+    def _fetch_data(self):
+        """Fetch latest data from WazeRouteCalculator."""
+        import WazeRouteCalculator
+        _LOGGER.debug("Update in progress")
+        try:
+            params = WazeRouteCalculator.WazeRouteCalculator(
+                self._origin, self._destination, self._region, None)
+            results = params.calc_all_routes_info()
+            best_route = next(iter(results))
```

Comment:
`results` is a dict - and the first item of a dict doesn't have to have the lowest-value/first-added key (only in Python 3.6).